auth/src/test/kotlin/com/tidal/sdk/auth/storage/DefaultTokensStoreMigrationsTest.kt

```diff
@@ -0,0 +1,107 @@
+package com.tidal.sdk.auth.storage
+
+import android.content.SharedPreferences
+import com.tidal.sdk.auth.model.Tokens
+import com.tidal.sdk.auth.storage.legacycredentials.LegacyCredentialsV1
+import com.tidal.sdk.auth.storage.legacycredentials.LegacyCredentialsV2
+import com.tidal.sdk.auth.storage.legacycredentials.Scopes
+import com.tidal.sdk.auth.storage.legacycredentials.TokensV1
+import com.tidal.sdk.auth.storage.legacycredentials.TokensV2
+import io.mockk.every
+import io.mockk.mockk
+import junit.framework.TestCase.assertEquals
+import kotlin.test.Test
+import kotlinx.datetime.Instant
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.junit.jupiter.api.BeforeEach
+
+class DefaultTokensStoreMigrationsTest {
+    private val testCredentialsKey = "testKey"
+    private var isPrefsEmpty = true
+    private lateinit var sharedPreferences: SharedPreferences
+    private lateinit var editor: SharedPreferences.Editor
+    private lateinit var defaultTokensStore: DefaultTokensStore
+
+    @BeforeEach
+    fun setUp() {
+        sharedPreferences = mockk()
+        editor = mockk(relaxed = true)
+
+        every { sharedPreferences.edit() } returns editor
+
+        defaultTokensStore = DefaultTokensStore(testCredentialsKey, sharedPreferences)
+    }
+
+    @Test
+    fun `getLatestTokens saves and returns Tokens when SharedPrefs returns LegacyCredentialsV1`() {
+        // given
+        isPrefsEmpty = false
+        val scopes = Scopes.fromString(setOf("scope1", "scope2").joinToString(""))
+        val legacyCredentialsV1 = LegacyCredentialsV1(
+            clientId = "clientId",
+            requestedScopes = scopes,
+            clientUniqueKey = "clientUniqueKey",
+            grantedScopes = scopes,
+            userId = "userId",
+            expires = null,
+            token = "token",
+        )
+        val refreshToken = "refreshToken"
+        val polymorphicTokens = TokensV1(legacyCredentialsV1, refreshToken)
+        val expected = Tokens(legacyCredentialsV1.toCredentials(), refreshToken)
+
+        every { sharedPreferences.getString(any(), any()) } answers {
+            if (isPrefsEmpty) {
```

Comment:
Nitpick: This if-else statement and the `isPrefsEmpty` flag seem to be unnecessary. The flag is never set to true, and in these tests, we always expect sharedPreferences to return something

Reply:
This is true, let's throw em out